Bottom: 3714a8d9b088e67f4ed134a3b75bd33449f34c0e
Top:    d03ba9ad43805a0f0e89ef111dcf7a3773dbcaf6
Author: Jason Wang <jasowang@redhat.com>
Date:   2017-11-14 19:24:00 +0800

Refresh of tuntap-ndo_xdp_xmit-support

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 9e34833..930b349 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1207,6 +1207,7 @@ static int tun_xdp_xmit(struct net_device *dev, struct xdp_buff *xdp)
 	/* Encode the XDP flag into lowest bit for consumer to differ
 	 * XDP buffer from sk_buff. */
 	if (ptr_ring_produce(ring, tun_xdp_to_ptr(buff))) {
+		/* FIXME: seems unnecessary */
 		/* Notify and wake up reader process */
 		if (tfile->flags & TUN_FASYNC)
 			kill_fasync(&tfile->fasync, SIGIO, POLL_IN);
@@ -1992,8 +1993,7 @@ static ssize_t tun_put_user(struct tun_struct *tun,
 	return total;
 }
 
-static void *tun_ring_recv(struct tun_file *tfile, int noblock,
-			   int *err)
+static void *tun_ring_recv(struct tun_file *tfile, int noblock, int *err)
 {
 	DECLARE_WAITQUEUE(wait, current);
 	void *ptr = NULL;
